src/explorer/databaseTreeItem.ts

```diff
@@ -53,7 +55,24 @@ export default class DatabaseTreeItem extends vscode.TreeItem
       return Promise.resolve([]);
     }
 
-    if (this._childrenCacheIsUpToDate) {
+    if (this.cacheIsUpToDate) {
+      const pastChildrenCache = this._childrenCache;
+      this._childrenCache = {};
+
+      // We manually rebuild each node to ensure we update the expanded state.
+      Object.keys(pastChildrenCache).forEach(collectionName => {
+        this._childrenCache[collectionName] = new CollectionTreeItem(
+          pastChildrenCache[collectionName].collection,
+          this.databaseName,
+          this._dataService,
+          pastChildrenCache[collectionName].isExpanded,
+          pastChildrenCache[collectionName].cacheIsUpToDate,
+          pastChildrenCache[collectionName].getDocumentListChild(),
+          pastChildrenCache[collectionName].getSchemaChild(),
+          pastChildrenCache[collectionName].getIndexListChild()
+        );
+      });
+
       return Promise.resolve(Object.values(this._childrenCache));
```

Comment:
Why not using `async/await`? What happens if this function throws synchronously, would we need a `try / catch`?

Reply:
Ah async await will be better. Easier to read and handle errors. Coming soon to a commit near you